src/modules/pages/NewApiPage/NewApiForm.js

```diff
@@ -21,272 +23,343 @@ import MultiSelect from '../../selects/MultiSelect/MultiSelect';
-                                        component={Radio}
-                                        value={'true'}
-                                        type="radio"
-                                        id="is-active"
-                                    />
-                                    <Label htmlFor="is-active">Yes</Label>
-                                </Row>
-                                <Row className={b('radio')()}>
-                                    <Field
-                                        name="active"
-                                        component={Radio}
-                                        value={'false'}
-                                        type="radio"
-                                        id="is-not-active"
-                                    />
-                                    <Label htmlFor="is-not-active">No</Label>
+    handleChangeStrategy = value => {
+        this.setState(() => ({
+            upstreams: {
+                balancing: value.value,
+                targets: value.options,
+            }
+        }), () => this.props.change('proxy.upstreams.balancing', value.value));
+    };
+
+    renderRoundrobinTargets = () => (
+        <div className={row({fullwidth: true}).mix('j-api-form__row')}>
+            <Row className={b('row')()} fullwidth>
+                <RoundrobinTargets
+                    name="proxy.upstreams.targets"
+                    title="Roundrobin targets"
+                />
+            </Row>
+        </div>
+    );
+
+    renderWeightTargets = () => (
+        <div className={row({fullwidth: true}).mix('j-api-form__row')}>
+            <Row className={b('row')()} fullwidth>
+                <WeightTargets
+                    name="proxy.upstreams.targets"
+                    title="Weight targets"
+                />
+            </Row>
+        </div>
+    );
+
+    renderStrategy = balancing => {
+        switch (balancing) {
+            case 'roundrobin': {
+                return this.renderRoundrobinTargets();
+            }
+            case 'weight': {
+                return this.renderWeightTargets();
+            }
+            default:
+                return null;
+        }
+    };
+
+    render() {
+        const {
+            handleSubmit,
+            initialValues,
+            apiSchema,
+            plugins,
+            excludePlugin,
+            selectPlugin,
+            selectedPlugins,
+        } = this.props;
+        // console.error('api', apiSchema);
+
+        const parse = value => value && parseInt(value);
+        const includePlugin = value => {
+            apiSchema.plugins
+                .filter((plugin, index) => {
+                    return plugin.name === value.value && !selectedPlugins.includes(plugin.name);
+                })
+                .map((plugin, index) => {
+                    selectPlugin(plugin.name);
+                });
+        };
+        const removePlugin = value => {
```

Comment:
two function with the same functionally might be hard to reason about, maybe you could find a better name to it and use in both scenarios